server/enterprise/message_export/message_export_test_e2e_generators.go

```diff
@@ -22,6 +22,104 @@ import (
 	"github.com/mattermost/mattermost/server/v8/platform/shared/filestore"
 )
 
+type MessageExport struct {
+	TeamId          string
+	TeamName        string
+	TeamDisplayName string
+
+	ChannelId          string
+	ChannelName        string
+	ChannelDisplayName string
+	ChannelType        model.ChannelType
+
+	UserId    string
+	UserEmail string
+	Username  string
+	IsBot     bool
+
+	PostId         string
+	PostCreateAt   int64
+	PostUpdateAt   int64
+	PostDeleteAt   int64
+	PostEditAt     int64
+	PostMessage    string
+	PostType       string
+	PostRootId     string
+	PostProps      string
+	PostOriginalId string
+	PostFileIds    []string
+}
+
+func removePointers(p *model.MessageExport) MessageExport {
+	return MessageExport{
+		TeamId:             model.SafeDereference(p.TeamId),
+		TeamName:           model.SafeDereference(p.TeamName),
+		TeamDisplayName:    model.SafeDereference(p.TeamDisplayName),
+		ChannelId:          model.SafeDereference(p.ChannelId),
+		ChannelName:        model.SafeDereference(p.ChannelName),
+		ChannelDisplayName: model.SafeDereference(p.ChannelDisplayName),
+		ChannelType:        model.SafeDereference(p.ChannelType),
+		UserId:             model.SafeDereference(p.UserId),
+		UserEmail:          model.SafeDereference(p.UserEmail),
+		Username:           model.SafeDereference(p.Username),
+		IsBot:              p.IsBot,
+		PostId:             model.SafeDereference(p.PostId),
+		PostCreateAt:       model.SafeDereference(p.PostCreateAt),
+		PostUpdateAt:       model.SafeDereference(p.PostUpdateAt),
+		PostDeleteAt:       model.SafeDereference(p.PostDeleteAt),
+		PostEditAt:         model.SafeDereference(p.PostEditAt),
+		PostMessage:        model.SafeDereference(p.PostMessage),
+		PostType:           model.SafeDereference(p.PostType),
+		PostRootId:         model.SafeDereference(p.PostRootId),
+		PostProps:          model.SafeDereference(p.PostProps),
+		PostOriginalId:     model.SafeDereference(p.PostOriginalId),
+		PostFileIds:        p.PostFileIds,
+	}
+}
+
+func waitUntilZeroPosts(t *testing.T, th *api4.TestHelper) {
+	// This is bananas -- sometimes there's a post in the future before we start the test, and it's causing failures.
+	for x := 0; x < 5; x++ {
+		count, err := th.App.Srv().Store().Post().AnalyticsPostCount(&model.PostCountOptions{ExcludeSystemPosts: true, SincePostID: "", SinceUpdateAt: model.GetMillis()})
+		require.NoError(t, err)
+		if count != 0 {
+			time.Sleep(1 * time.Second)
+		} else {
+			return
+		}
+	}
+	assert.Fail(t, "e2e setup failed", "waited 5 seconds and there was still a message in the future. Something is wrong.")
+}
```

Comment:
Is this something we consider retaining or did it serve its purpose already to ensure stability?

Reply:
We can remove it, it looks like the stability changes solved it.